security/basic/src/test/java/io/quarkus/ts/openshift/security/basic/SecurityTest.java

```diff
@@ -0,0 +1,49 @@
+package io.quarkus.ts.openshift.security.basic;
+
+import static io.restassured.RestAssured.given;
+import static org.hamcrest.CoreMatchers.equalTo;
+
+import org.apache.http.HttpStatus;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+
+import io.quarkus.test.junit.QuarkusTest;
+import io.quarkus.test.junit.TestProfile;
+import io.quarkus.test.security.TestSecurity;
+
+@Tag("QUARKUS-1584")
+@QuarkusTest
+@TestProfile(SecurityTestProfile.class)
+public class SecurityTest {
+
+    @Test
+    public void permitAll() {
+        given().get("/permit-all")
+                .then()
+                .statusCode(HttpStatus.SC_OK)
+                .body(equalTo("Hello!"));
+    }
+
+    @Test
+    public void denyAll() {
+        given().get("/deny-all")
+                .then()
+                .statusCode(HttpStatus.SC_UNAUTHORIZED);
+    }
+
+    @Test
+    public void denyUnannotated() {
+        given().get("/unannotated")
+                .then()
+                .statusCode(HttpStatus.SC_UNAUTHORIZED);
+    }
+
+    @Test
+    @TestSecurity(authorizationEnabled = false)
+    public void allowUnannotated() {
+        given().get("/unannotated")
+                .then()
+                .statusCode(HttpStatus.SC_OK)
+                .body(equalTo("Hello!"));
+    }
```

Comment:
I am missing an scenario that cover this feature:

```
@TestSecurity(user = "isaac", roles = { "user" })
```

Basically, we need to add another test that validates that we are setting a user and a role by the above annotation. 

The test could be something like:
```
@Test
    @TestSecurity(user = "testUser", roles = { "user" })
    public void getUser() {
        given().get("/user")
                .then()
                .statusCode(HttpStatus.SC_OK)
                .body(containsString("testUser"));
    }
```
and / or

```
@Test
    @TestSecurity(user = "testUser", roles = { "admin" })
    public void getUser() {
        given().get("/user")
                .then()
                .statusCode(HttpStatus.SC_FORBIDDEN);
    }
```